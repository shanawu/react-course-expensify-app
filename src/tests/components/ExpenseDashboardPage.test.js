import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "../../store/configureStore";
import ExpenseDashboardPage from "../../components/ExpenseDashboardPage";
import "react-dates/initialize";

test("should render Expense Dashboard Page correctly", () => {
  const store = configureStore();
  const { container } = render(
    <Provider store={store}>
      <ExpenseDashboardPage />
    </Provider>
  );

  expect(container.firstChild).toMatchSnapshot();
});
