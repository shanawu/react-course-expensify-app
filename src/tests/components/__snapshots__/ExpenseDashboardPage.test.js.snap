// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense Dashboard Page correctly 1`] = `
<div>
  <div>
    <input
      type="text"
      value=""
    />
    <select>
      <option
        value="date"
      >
        Date
      </option>
      <option
        value="amount"
      >
        Amount
      </option>
    </select>
    <div
      class="DateRangePicker DateRangePicker_1"
    >
      <div>
        <div
          class="DateRangePickerInput DateRangePickerInput_1 DateRangePickerInput__withBorder DateRangePickerInput__withBorder_2 DateRangePickerInput__showClearDates DateRangePickerInput__showClearDates_3"
        >
          <div
            class="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-startDate"
              aria-label="Start Date"
              autocomplete="off"
              class="DateInput_input DateInput_input_1"
              id="startDate"
              name="startDate"
              placeholder="Start Date"
              type="text"
              value="07/01/2023"
            />
            <p
              class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-startDate"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <div
            aria-hidden="true"
            class="DateRangePickerInput_arrow DateRangePickerInput_arrow_1"
            role="presentation"
          >
            <svg
              class="DateRangePickerInput_arrow_svg DateRangePickerInput_arrow_svg_1"
              focusable="false"
              viewBox="0 0 1000 1000"
            >
              <path
                d="M694 242l249 250c12 11 12 21 1 32L694 773c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210-210H68c-13 0-23-10-23-23s10-23 23-23h806L662 275c-21-22 11-54 32-33z"
              />
            </svg>
          </div>
          <div
            class="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-endDate"
              aria-label="End Date"
              autocomplete="off"
              class="DateInput_input DateInput_input_1"
              id="endDate"
              name="endDate"
              placeholder="End Date"
              type="text"
              value="07/31/2023"
            />
            <p
              class="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-endDate"
            >
              Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
          <button
            aria-label="Clear Dates"
            class="DateRangePickerInput_clearDates DateRangePickerInput_clearDates_1 DateRangePickerInput_clearDates_default DateRangePickerInput_clearDates_default_2"
            type="button"
          >
            <svg
              class="DateRangePickerInput_clearDates_svg DateRangePickerInput_clearDates_svg_1"
              focusable="false"
              viewBox="0 0 12 12"
            >
              <path
                d="M11.53.47a.75.75 0 0 0-1.061 0l-4.47 4.47L1.529.47A.75.75 0 1 0 .468 1.531l4.47 4.47-4.47 4.47a.75.75 0 1 0 1.061 1.061l4.47-4.47 4.47 4.47a.75.75 0 1 0 1.061-1.061l-4.47-4.47 4.47-4.47a.75.75 0 0 0 0-1.061z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div>
    <h1>
      Expense List
    </h1>
    <p>
      No expenses
    </p>
  </div>
</div>
`;
